pkg/auth/providers/saml/saml_provider.go

```diff
@@ -223,17 +302,31 @@ func (s *Provider) toPrincipal(principalType string, princ v3.Principal, token *
 }
 
 func (s *Provider) RefetchGroupPrincipals(principalID string, secret string) ([]v3.Principal, error) {
-	// This should never be called
+	if s.hasLdapGroupSearch() {
+		externalID, principalType := splitPrincipalID(principalID)
+		if principalType == "user" {
+			userPrincipals, err := s.ldapProvider.SearchPrincipals(externalID, "user", v3.Token{})
+			if err != nil || len(userPrincipals) != 1 {
+				return nil, err
+			}
+
+			return s.ldapProvider.RefetchGroupPrincipals(userPrincipals[0].Name, secret)
+		}
+	}
+
 	return nil, errors.New("Not implemented")
 }
 
 func (s *Provider) SearchPrincipals(searchKey, principalType string, token v3.Token) ([]v3.Principal, error) {
 	var principals []v3.Principal
-
 	if principalType == "" {
 		principalType = "user"
 	}
 
+	if principalType == s.groupType && s.hasLdapGroupSearch() {
```

Comment:
We should call search principal for users too, till now since we couldn't perform search for saml we have been adding anything user types in as a user